src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -264,20 +264,64 @@
                :message (str "Invalid service_status_version. Should be an "
                           "integer but was " level)}))))
 
+(schema/defn ^:always-validate summarize-states
+  "Given a map of service statuses:
+   * if all of the statuses have the same :state, returns that :state
+   * if not all of the statuses are the same, returns :unknown"
+  [statuses :- ServicesStatus] :- State
+  (let [state-set (->> statuses
+                       vals
+                       (map :state)
+                       set)]
+    (if (= 1 (count state-set))
+      (first state-set)
+      :unknown)))
```

Comment:
I don't think this is the right behavior. If some of the services are in an :error state, but some are :running, I think the summary should be that the whole system is in an :error state. If none of them are in :error, but at least one is :unknown, then I think it makes sense to report :unknown. So basically, the ordering is :error > :unknown > :running, and the summary behavior is to choose the greatest from among the reported states by that ordering.